File: backend/ArduinoThermoHygrometer/DTOs/BatteryDto.cs
namespace ArduinoThermoHygrometer.Web.DTOs;

public record BatteryDto
{
    public Guid Id { get; set; } = default!;

    public DateTimeOffset RegisteredAt { get; set; } = default!;

    public int BatteryStatus { get; set; } = default!;
}

File: api/src/main/java/api/arduinothermohygrometer/dtos/BatteryDto.java
package api.arduinothermohygrometer.dtos;

import java.time.LocalDateTime;

import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.Builder;

@Builder
public record BatteryDto(@NotNull LocalDateTime registeredAt,
                         @Min(0) @Max(100) int batteryStatus
) {}
